packages/block-editor/src/components/block-list/index.js

```diff
@@ -318,22 +60,65 @@ export default compose( [
 				! isTyping() &&
 				getGlobalBlockCount() <= BLOCK_ANIMATION_THRESHOLD
 			),
-			getBlockParents,
 		};
-	} ),
-	withDispatch( ( dispatch ) => {
-		const {
-			startMultiSelect,
-			stopMultiSelect,
-			multiSelect,
-			selectBlock,
-		} = dispatch( 'core/block-editor' );
+	}
 
-		return {
-			onStartMultiSelect: startMultiSelect,
-			onStopMultiSelect: stopMultiSelect,
-			onMultiSelect: multiSelect,
-			onSelectBlock: selectBlock,
-		};
-	} ),
-] )( BlockList );
+	const {
+		blockClientIds,
+		isMultiSelecting,
+		selectedBlockClientId,
+		multiSelectedBlockClientIds,
+		hasMultiSelection,
+		enableAnimation,
+	} = useSelect( selector );
+	const ref = useRef();
+	const onSelectionStart = useSelection( { ref, rootClientId } );
```

Comment:
I like the extraction :) I believe it should "mutliSelection" and not just "selection" to avoid confusion.

Reply:
I think it would be good to check performance impact (new useSelect...)

Reply:
Done :)

Reply:
Oh, right. I'll run the tests.

Reply:
Seems to be more or less the same.

This branch:

```
Average time to load: 4447ms
Average time to DOM content load: 4163ms
Average time to type character: 68.645ms
Slowest time to type character: 116ms
Fastest time to type character: 56ms
```

`master`:

```
Average time to load: 4510ms
Average time to DOM content load: 4224ms
Average time to type character: 69.04ms
Slowest time to type character: 118ms
Fastest time to type character: 50ms
```